// react-snipcart/cart-has-qty.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _dirtyCheckCartQty = require('./dirty-check-cart-qty');

var _dirtyCheckCartQty2 = _interopRequireDefault(_dirtyCheckCartQty);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var CartHasQty = function (_React$Component) {
	_inherits(CartHasQty, _React$Component);

	function CartHasQty(props) {
		_classCallCheck(this, CartHasQty);

		var _this = _possibleConstructorReturn(this, (CartHasQty.__proto__ || Object.getPrototypeOf(CartHasQty)).call(this, props));

		_this.state = {
			hasQty: false
		};
		_this.onChange = _this.onChange.bind(_this);
		return _this;
	}

	_createClass(CartHasQty, [{
		key: 'onChange',
		value: function onChange(qty) {
			var hasQty = qty ? true : false;
			if (hasQty !== this.state.hasQty) {
				this.setState({
					hasQty: hasQty
				});
			}
		}
	}, {
		key: 'componentDidMount',
		value: function componentDidMount() {
			_dirtyCheckCartQty2.default.addChangeEvent(this.onChange);
		}
	}, {
		key: 'componentWillUnmount',
		value: function componentWillUnmount() {
			_dirtyCheckCartQty2.default.removeChangeEvent(this.onChange);
		}
	}, {
		key: 'render',
		value: function render() {
			return _react2.default.createElement(
				'div',
				null,
				this.state.hasQty ? this.props.children : this.props.noQty
			);
		}
	}]);

	return CartHasQty;
}(_react2.default.Component);

exports.default = CartHasQty;
